packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1063,16 +1304,136 @@ export class TransactionController extends BaseControllerV1<
     this.hub.emit(`${transactionMeta.id}:speedup`, newTransactionMeta);
   }
 
+  // NOTE(JL): Should this be private?
+  stopTrackingByNetworkClientId(networkClientId: NetworkClientId) {
+    const trackers = this.trackingMap.get(networkClientId);
+    if (trackers) {
+      trackers.pendingTransactionTracker.stop();
+      this.removePendingTransactionTrackerListeners(
+        trackers.pendingTransactionTracker,
+      );
+      trackers.incomingTransactionHelper.stop();
+      this.removeIncomingTransactionHelperListeners(
+        trackers.incomingTransactionHelper,
+      );
+      this.trackingMap.delete(networkClientId);
+    }
+  }
```

Comment:
should revisit this

Reply:
I can look into this

Reply:
https://github.com/MetaMask/core/pull/3860